### PR TITLE
[Python][Choice] Add recognizers-text-choice to Python implementation

### PR DESCRIPTION
**Pull request by Aliandi**
### Description
This PR adds the support of `recognizers-text-choice` for Python. To the moment, the only ported language is **English**.

### Changes made

- Add all base classes for _extractor_, _parser_ and _model_
- Make tests for Choice available on Python
- Modify the Choice English patterns to use the correct _regex_ for emojis on Python
- Add support for English Choice 
- Add [regex](https://pypi.org/project/regex/), [grapheme](https://pypi.org/project/grapheme/) and [emoji](https://pypi.org/project/emoji/) libraries for Python
### Testing
_Running tests:_

imagen

### CODE REVIEW

File: .NET/Microsoft.Recognizers.Definitions.Common/English/DateTimeDefinitions.cs [hunk 1]
```diff
@@ -19,749 +19,733 @@ namespace Microsoft.Recognizers.Definitions.English
...
-      public static readonly IList<string> GenericYearTerms = new List<string>
-        {
-            @"y"
-        };
-      public static readonly IList<string> YearToDateTerms = new List<string>
-        {
-            @"year to date"
-        };
-    }
+	public static class DateTimeDefinitions
```
Comment (gasper-az) [inline .NET/Microsoft.Recognizers.Definitions.Common/English/DateTimeDefinitions.cs:22]: Why is this `C#` file in this `python` PR??

File: Python/libraries/recognizers-choice/recognizers_choice/choice/constants.py~HEAD [hunk 2]
```diff
@@ -0,0 +1,4 @@
+class Constants:
```
Comment (gasper-az) [inline Python/libraries/recognizers-choice/recognizers_choice/choice/constants.py~HEAD:1]: This file shouldn't be here

File: Python/libraries/recognizers-choice/recognizers_choice/choice/english/boolean.py [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+class EnglishBooleanExtractorConfiguration(BooleanExtractorConfiguration):
+    regex_true: Pattern
+    regex_false: Pattern
+    token_regex: Pattern
+    only_top_match: bool
+
+    def __init__(self, only_top_match: bool = True):
+        self.regex_true = RegExpUtility.get_safe_reg_exp(EnglishChoice.TrueRegex)
+        self.regex_false = RegExpUtility.get_safe_reg_exp(EnglishChoice.FalseRegex)
+        self.token_regex = RegExpUtility.get_safe_reg_exp(EnglishChoice.TokenizerRegex, regex.S)
```
Comment (gasper-az) [inline Python/libraries/recognizers-choice/recognizers_choice/choice/english/boolean.py:18]: is the `regex.S` name ok??

File: JavaScript/packages/recognizers-choice/src/resources/chineseChoice.ts [hunk 4]
```diff
@@ -4,14 +4,11 @@
...
-// Licensed under the MIT License.
-// ------------------------------------------------------------------------------
-
-export namespace ChineseChoice {
-    export const LangMarker = 'Chs';
-    export const TokenizerRegex = `[^\\u3040-\\u30ff\\u3400-\\u4dbf\\u4e00-\\u9fff\\uf900-\\ufaff\\uff66-\\uff9f]`;
-    export const TrueRegex = `(好[的啊呀嘞哇]|没问题|可以|中|好|同意|行|是的|是|对)|(\\uD83D\\uDC4D|\\uD83D\\uDC4C)`;
-    export const FalseRegex = `(不行|不好|拒绝|否定|不中|不可以|不是的|不是|不对|不)|(\\uD83D\\uDC4E|\\u270B|\\uD83D\\uDD90)`;
-}
+// ------------------------------------------------------------------------------
```
Comment (gasper-az) [inline JavaScript/packages/recognizers-choice/src/resources/chineseChoice.ts:7]: Same here. Why are we introducing `typescript` files?

**Review by gasper-az — COMMENTED** [review]
Just some minor feedback

File: Python/libraries/recognizers-choice/recognizers_choice/choice/english/boolean.py (hunk 3 above)
Comment (Aliandi) [reply Python/libraries/recognizers-choice/recognizers_choice/choice/english/boolean.py:18]: Yes

File: JavaScript/packages/recognizers-choice/src/resources/chineseChoice.ts (hunk 4 above)
Comment (Aliandi) [reply JavaScript/packages/recognizers-choice/src/resources/chineseChoice.ts:7]: Removed it!

File: Python/libraries/recognizers-choice/recognizers_choice/choice/constants.py~HEAD (hunk 2 above)
Comment (Aliandi) [reply Python/libraries/recognizers-choice/recognizers_choice/choice/constants.py~HEAD:1]: removed it

**Pull request closed without merging**